crates/bevy_ui/src/widget/text.rs

```diff
@@ -219,49 +235,47 @@ fn queue_text(
 #[allow(clippy::too_many_arguments)]
 pub fn text_system(
     mut textures: ResMut<Assets<Image>>,
-    mut last_scale_factor: Local<f32>,
+    mut last_scale_factors: Local<EntityHashMap<f32>>,
     fonts: Res<Assets<Font>>,
-    windows: Query<&Window, With<PrimaryWindow>>,
+    camera_query: Query<(Entity, &Camera)>,
+    default_ui_camera: DefaultUiCamera,
     text_settings: Res<TextSettings>,
     ui_scale: Res<UiScale>,
     mut texture_atlases: ResMut<Assets<TextureAtlasLayout>>,
     mut font_atlas_sets: ResMut<FontAtlasSets>,
     mut text_pipeline: ResMut<TextPipeline>,
-    mut text_query: Query<(Ref<Node>, &Text, &mut TextLayoutInfo, &mut TextFlags)>,
+    mut text_query: Query<(
+        Ref<Node>,
+        &Text,
+        &mut TextLayoutInfo,
+        &mut TextFlags,
+        Option<&TargetCamera>,
+    )>,
 ) {
-    // TODO: Support window-independent scaling: https://github.com/bevyengine/bevy/issues/5621
-    let window_scale_factor = windows
-        .get_single()
-        .map(|window| window.resolution.scale_factor())
-        .unwrap_or(1.);
+    let mut scale_factors: EntityHashMap<f32> = EntityHashMap::default();
```

Comment:
```suggestion
    // PERF: this could be cached in a Local as well to save a recurring allocation
    let mut scale_factors: EntityHashMap<f32> = EntityHashMap::default();
```

Reply:
no, this one can't be in a local, it's the one used to check if the local-saved one changed.

Reply:
though that cache of scale factor (and size for other UI parts) per camera could be computed once instead of several times for each UI component

Reply:
My thought was that you cache it and then clear it every frame :)